sub_peers.js

```diff
@@ -159,7 +240,9 @@ TChannelSubPeers.prototype.chooseHeapPeer = function chooseHeapPeer(req) {
     var self = this;
 
     var peer;
-    if (req && req.triedRemoteAddrs) {
+    if ((req && req.triedRemoteAddrs) ||
+        (this.hasMinConnections && this.currentConnectedPeers < this.minConnections)
```

Comment:
Seems like the second half of this or could be brought out and calculated once:
```
var notEnoughPeers = self.hasMinConnections && this.currentConnectedPeers < minConnections
```
Then you could use it in the filter function blow as well